core/templates/components/score-ring/score-ring.component.ts

```diff
@@ -67,13 +76,45 @@ export class ScoreRingComponent implements OnInit, OnChanges {
   }
 
   ngOnInit(): void {
-    this.circle = (
-      document.querySelector('.score-ring-circle') as SVGCircleElement);
+    // In certain case(s), such as when the results page is reloaded,
+    // document.querySelector() attempts to grab the required element before
+    // it even exists in the view causing it to return null,
+    // thus breaking the score-ring. Checking for the existence of
+    // the element before retrieving it fixes this problem.
+    // ngOnInit() (and hence getScoreRing() too) is called recursively until
+    // the circle element is returned by document.querySelector()
+
+    if (!this.getScoreRing()) {
+      let that = this;
+      setTimeout(function() {
+        that.ngOnInit();
+      }, 160);
+      return;
+    }
```

Comment:
Can we use ngAfterViewInit() to solve this issue?

Reply:
@Rijuta-s I have tried making use of `ngAfterViewInit()`, however I have not been able to get it to work (unless I'm missing something).
I have imported it, implemented it in the class declaration, and the function body looks as follows:
![image](https://user-images.githubusercontent.com/78612244/146636896-952f34e5-d706-45fd-99dc-fe6013cba857.png)

'element doesn't exist yet' gets logged in the console.

Alternatively, if the concern here is that multiple `ngOnInit()` calls shouldn't be made, I could change the code so it looks as follows:

-- All the code currently in the `ngOnInit()` function body (used for initializing the score-ring) can be moved into a seperate function (`intializeScoreRing()`).

-- This new function will be called exactly once in `ngOnInit()`.

-- The new function will have a check to see if `document.querySelector()` finds the required element. If the element doesn't exist yet, `initializeScoreRing` will call itself recursively using setTimeout() until `document.querySelector()` stops returning null.

This is the same as the current code changes in the PR, except it is not `ngOnInit()` that will be called multiple times, it's a seperate function within `ngOnInit()`. `ngOnInit()` itself will only be called once.

Could you let me know if you can find anything wrong with the use of `ngAfterViewInit()` in my code, or if the second solution is an acceptable alternative?

Reply:
@Rijuta-s I have tried making use of `ngAfterViewInit()`, but it doesn't seem to make a difference (unless I'm missing something).
I've imported it, implemented it in the class declaration, and the function body looks as follows:
![image](https://user-images.githubusercontent.com/78612244/146638998-3ca4dfe2-635e-46b3-bd2c-0c94a01c5026.png)

'element doesn't exist yet' gets logged in the console.

Alternatively, if the concern is that `ngOnInit()` shouldn't be called multiple times, I could make the following changes:

-- All the code inside `ngOnInit()` (used for initializing the score-ring) can be moved into a separate function (`initializeScoreRing()`).

-- `initializeScoreRing()` is called a single time in `ngOnInit()`.

-- `initializeScoreRing()` first checks to see if the element exists yet. If not, it will call itself in a `setTimeout` call and immediately return to stop the rest of the function. It will keep calling itself until the element exists and is returned by document.querySelector, upon which the rest of the function will be executed.

This is similar to the current changes in the PR, except it's not `ngOnInit()` that will be called multiple times, it's `initializeScoreRing()`. `ngOnInit()` itself will only be called once (i.e. upon component initialization).

Could you let me know if you find something I might be doing wrong with `ngAfterViewInit()` in my code, or if the second solution is an acceptable alternative?

Reply:
Just because `document.querySelector` attempts to find an element in DOM before angular has rendered that element, we prefer to use template reference variables for these cases. I suggest using a template reference variable here instead and accessing it using the `ngAfterViewInit` hook.

Reply:
@ashutoshc8101 Thanks for this comment! I wasn't familiar with the concept of template variables, but I've looked into it, and your solution worked! Thanks a lot!

Reply:
Thanks @ashutoshc8101!

Reply:
Sorry, not yet! I have run into a small error, and have left a comment about the same. I have tagged Aishwary since he used a similar solution in his code so he may be able to tell me how to proceed.

Reply:
It appears Aishwary is busy as he hasn't been able to respond, so I've gone ahead and pushed the changes.

Reply:
@Rijuta-s I've updated the PR, PTAL!

Reply:
Thanks @nithinrdy LGTM!